src/mem/queue.rs

```diff
@@ -15,6 +15,7 @@ pub struct MemQueue {
     concatenated_records: Vec<u8>,
     start_position: u64,
     record_metas: Vec<RecordMeta>,
+    used_files: BTreeMap<RecordMeta, FileNumber>,
```

Comment:
I would have taken the position or the index in the record_meta Vec as a key in the BTreeMap... but why let's see.

Reply:
record_meta elements get reindexed when truncating, so this would mean extracting and reinserting every element on truncation, unless you had something else in mind?

Reply:
But upon truncation RecordMeta are also be updated aren't they?

Reply:
ho well, my changes aren't correct then, they don't update the key as they should 😟